tests/suites/0_stateless/05_0006_ddl_grant_privilege.sql

```diff
@@ -15,6 +15,8 @@ GRANT SELECT ON `db01`.'tb1' TO 'test-grant'@'localhost';
 GRANT SELECT ON db01.tbnotexists TO 'test-grant'@'localhost'; -- {ErrorCode 25}
 GRANT SELECT ON dbnotexists.* TO 'test-grant'@'localhost'; -- {ErrorCode 3}
 
+SHOW GRANTS FOR 'test-grant'@'localhost';
+
 REVOKE SELECT ON db01.* FROM 'test-grant'@'localhost';
```

Comment:
It would be better to check the privileges again after REVOKE:
SHOW GRANTS FOR 'test-grant'@'localhost';

Reply:
thank you, i've add two show grant statements after each revoke statement

Reply:
After `REVOKE SELECT ON db01.* FROM 'test-grant'@'localhost';`
Seems like the grants result is not correct?

Reply:
after revoke SELECT ON db01.* FROM 'test-grant'@'localhost', the rest grants is

```
GRANT CREATE,SELECT,INSERT,SET ON 'default'.* TO 'test-grant'@'localhost'
GRANT SELECT ON 'db01'.'tb1' TO 'test-grant'@'localhost'
```

the privilege on 'db01'.'tb1' is not revoked by the statement `REVOKE .. ON 'db01'.'*'`, the behaviour is counter-intuitive, but the same as mysql:

```
╰─$ mysql -uroot --port 3306 -hlocalhost --protocol=TCP                                                                     
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.35 Homebrew

Copyright (c) 2000, 2021, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> CREATE DATABASE IF NOT EXISTS `db01`;
Query OK, 1 row affected, 1 warning (0.01 sec)

mysql> CREATE TABLE IF NOT EXISTS `db01`.`tb1` (`id` NOT NULL);
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> CREATE USER 'test-grant'@'localhost' IDENTIFIED BY 'password';

mysql> GRANT SELECT ON db01.* TO 'test-grant'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT SELECT ON db01.tb1 TO 'test-grant'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> show grants for 'test-grant'@'localhost';
+-------------------------------------------------------------+
| Grants for test-grant@localhost                             |
+-------------------------------------------------------------+
| GRANT USAGE ON *.* TO 'test-grant'@'localhost'              |
| GRANT SELECT ON `dbnotexists`.* TO 'test-grant'@'localhost' |
| GRANT SELECT ON `db01`.* TO 'test-grant'@'localhost'        |
| GRANT SELECT ON `db01`.`tb1` TO 'test-grant'@'localhost'    |
+-------------------------------------------------------------+
4 rows in set (0.00 sec)

mysql> REVOKE SELECT ON db01.* FROM 'test-grant'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> show grants for 'test-grant'@'localhost';
+-------------------------------------------------------------+
| Grants for test-grant@localhost                             |
+-------------------------------------------------------------+
| GRANT USAGE ON *.* TO 'test-grant'@'localhost'              |
| GRANT SELECT ON `dbnotexists`.* TO 'test-grant'@'localhost' |
| GRANT SELECT ON `db01`.`tb1` TO 'test-grant'@'localhost'    |
+-------------------------------------------------------------+
3 rows in set (0.00 sec)
```

it seems grant & revoke statement matches the grant objects as key by key, but not taking the relationship between the grant objects into account. (db01.* does include db01.tb1, but revoking privileges on db01.* not revoking db01.tb1)